core/templates/pages/contributor-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -16,116 +16,117 @@
+}
 
-angular.module('oppia').factory('ContributionOpportunitiesService', [
-  '$uibModal', 'ContributionOpportunitiesBackendApiService',
-  'UrlInterpolationService',
-  function(
-      $uibModal, ContributionOpportunitiesBackendApiService,
-      UrlInterpolationService) {
-    var reloadOpportunitiesEventEmitter = new EventEmitter<void>();
-    var removeOpportunitiesEventEmitter = new EventEmitter<void>();
+class ExplorationOpportunitiesDict {
+  opportunities: ExplorationOpportunitySummary[];
+  more: boolean;
+}
 
-    var skillOpportunitiesCursor = null;
-    var translationOpportunitiesCursor = null;
-    var voiceoverOpportunitiesCursor = null;
-    var moreSkillOpportunitiesAvailable = true;
-    var moreTranslationOpportunitiesAvailable = true;
-    var moreVoiceoverOpportunitiesAvailable = true;
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributionOpportunitiesService {
+  constructor(
+    private contributionOpportunitiesBackendApiService:
+      ContributionOpportunitiesBackendApiService,
+      private modalService: NgbModal) {}
 
-    var _getSkillOpportunities = function(cursor) {
-      return ContributionOpportunitiesBackendApiService
-        .fetchSkillOpportunitiesAsync(
-          cursor).then(({ opportunities, nextCursor, more }) => {
-          skillOpportunitiesCursor = nextCursor;
-          moreSkillOpportunitiesAvailable = more;
-          return {
-            opportunities: opportunities,
-            more: more
-          };
-        });
-    };
-    var _getTranslationOpportunities = function(languageCode, cursor) {
-      return ContributionOpportunitiesBackendApiService
-        .fetchTranslationOpportunitiesAsync(
-          languageCode, cursor).then(({ opportunities, nextCursor, more }) => {
-          translationOpportunitiesCursor = nextCursor;
-          moreTranslationOpportunitiesAvailable = more;
-          return {
-            opportunities: opportunities,
-            more: more
-          };
-        });
-    };
-    var _getVoiceoverOpportunities = function(languageCode, cursor) {
-      return ContributionOpportunitiesBackendApiService
-        .fetchVoiceoverOpportunitiesAsync(languageCode, cursor).then(
-          function({ opportunities, nextCursor, more }) {
-            voiceoverOpportunitiesCursor = nextCursor;
-            moreVoiceoverOpportunitiesAvailable = more;
-            return {
-              opportunities: opportunities,
-              more: more
-            };
-          });
-    };
+  public reloadOpportunitiesEventEmitter = new EventEmitter<void>();
+  public removeOpportunitiesEventEmitter = new EventEmitter<void>();
+  private _skillOpportunitiesCursor: string = null;
+  private _translationOpportunitiesCursor: string = null;
+  private _voiceoverOpportunitiesCursor: string = null;
+  private _moreSkillOpportunitiesAvailable = true;
+  private _moreTranslationOpportunitiesAvailable = true;
+  private _moreVoiceoverOpportunitiesAvailable = true;
```

Comment:
Re earlier comment: Yup I think the event emitter type declarations can be omitted because of type inference. 

I also think we can drop the underscore prefixes here since they will be redundant with the `private` declarations.

Reply:
I'm following the underscore convention as per [this](https://github.com/oppia/oppia/pull/11409#discussion_r544784985) comment on one of my previous PRs.
I think we're sticking with the underscore prefix convention post migration, but this does seem redundant.
@srijanreddy98, would it make more sense to drop these underscores?
It's a little confusing, especially since the injected services aren't prefixed.

Reply:
We had a similar discussion related to interfaces and prefixes: https://github.com/oppia/oppia/issues/9348 and various leads talked about potentially moving away from this convention. It may be worth revisiting this and track it in a separate issue.